src/transaction.rs

```diff
@@ -134,6 +134,48 @@ impl<'a> Transaction<'a> {
         Ok(())
     }
 
+    fn _load_build_app_cache(&self, app: &crate::models::Application) -> Result<(), EngineError> {
+        // do load build cache before building app
+        let _ = match self.engine.build_platform().has_cache(app.to_build()) {
+            Ok(CacheResult::Hit) => {
+                info!("cache hit for app {}", app.name.as_str());
+            }
+            Ok(CacheResult::Miss(parent_build)) => {
+                info!("cache miss for app {}", app.name.as_str());
+
+                let container_registry = self.engine.container_registry();
+
+                // pull image from container registry
+                // FIXME: if one day we use something else than LocalDocker to build image
+                // FIXME: we'll need to send the result of the pull to the implementation of Build
+                let _ = match container_registry.pull(&parent_build.image) {
+                    Ok(pull_result) => pull_result,
+                    Err(err) => {
+                        warn!(
+                            "{}",
+                            err.message.clone().unwrap_or(format!(
+                                "something goes wrong while pulling image from {:?} container registry",
+                                container_registry.kind()
+                            ))
+                        );
+                        return Err(err);
+                    }
+                };
+            }
+            Err(err) => {
+                warn!(
+                    "load build app {} cache error: {}",
+                    app.name.as_str(),
+                    err.message.clone().unwrap_or("<no message>".to_string())
+                );
+
+                return Err(err);
+            }
+        };
+
+        Ok(())
+    }
+
     fn _build_applications(
```

Comment:
Same, why initial `_` ?

Reply:
it's a personal convention to say that it is an internal function. Not important, we can remove it. (It's a convention used in Python). Happy to remove it.

Reply:
Ok, well, I guess the fact it's private is already a good signal it's internal. If you don't mind, removing it would help adding some coherence.